ibis/expr/operations.py

```diff
@@ -1472,22 +1472,20 @@ class OuterJoin(Join):
     pass
 
 
-class LeftSemiJoin(Join):
+class InnerSemiJoin(Join):
 
-    """
+    def _get_schema(self):
+        return self.left.schema()
```

Comment:
In light of the clickhouse test suite the schema of this operation isn't known until you select columns out of it, similar to the rest of ibis's join operations.